src/item.cpp

```diff
@@ -5778,22 +5778,68 @@ bool item::process_food( player * /*carrier*/, const tripoint &pos )
-        item_counter -= diff_cold;
+    const auto temp = g->get_temperature( pos );
+    unsigned int diff_freeze = abs( temp - FREEZING_TEMPERATURE ) / 5 ;
+    diff_freeze = clamp( diff_freeze, static_cast<unsigned int>(0) , static_cast<unsigned int>(3) ); //effective 1-4
+
+    // div by 5 means every 5 degrees of difference equals 1 point of ratio
+    unsigned int diff_cold = abs( temp - FRIDGE_TEMPERATURE ) / 5;
+    diff_cold = clamp( diff_cold, static_cast<unsigned int>(0), static_cast<unsigned int>(3) ); //effective 1-4
+
+    // exclusions for items being actively cooled to be affected by outside temperatures
+    //const optional_vpart_position vp = g->m.veh_at( pos );
+    const optional_vpart_position vp = g->m.veh_at( pos );
+    bool in_active_cooler = false;
+    bool fridge_here = false;
+    bool freezer_here = false;
+    if( vp ) {
+        fridge_here = vp->vehicle().has_part( pos, "FRIDGE", true );
+        freezer_here = vp->vehicle().has_part( pos, "FREEZER", true );
+        if(fridge_here) { add_msg( "Fridge" ); } else {add_msg( "NOT Fridge" );}
+        if(freezer_here) { add_msg( "Freezer" ); } else {add_msg( "NOT Freezer" );}
+    }
+
+    if( vp && ( fridge_here || freezer_here ) ) {
+        std::vector<vehicle_part *> parts = vp->vehicle().get_parts( pos, "FRIDGE", true );
+        if( parts.empty() ) { 
+            parts = vp->vehicle().get_parts( pos, "FREEZER", true );
+        }
+        if( parts.empty() ) { add_msg( "PARTS EMPTY" ); }
+        vehicle_part *part = parts.front();
+        vehicle_stack vit = vp->vehicle().get_items( vp->vehicle().index_of_part( part ) );
+        if( vit.empty() ) { add_msg( "ITEMS EMPTY" ); }
+        for( size_t iter = 0; iter < vit.size(); iter++ ) {
+            item *itm = &vit[iter];
+            if( itm == this ) {
+                in_active_cooler = true;
+                add_msg( "PING" );	
+            } else { add_msg( "PONG %s", itm->tname() );	}
+        }
+    }
+    
+    // environment temperature applies COLD/FROZEN flags to food unless in fridge/freezer
+    if( !in_active_cooler ) {
+        if( temp <= FRIDGE_TEMPERATURE ) {
+            add_msg( "NOT_IN_COOLER" );	
+            g->m.apply_in_fridge( *this, temp );
+            add_msg( "Temp: %s", temp );	
+        } else if ( item_tags.count( "FROZEN" ) > 0 && item_counter > diff_freeze ) {
+            item_counter -= diff_freeze; // thaw
+        } else if( item_tags.count( "COLD" ) > 0 && item_counter > diff_cold ) {
+            item_counter -= diff_cold; // get warm
+        }
+    } else if( fridge_here ) {
+            add_msg( "IN_FRIDGE" );	
+        g->m.apply_in_fridge( *this, FRIDGE_TEMPERATURE );
+        add_msg( "Temp: %s", FRIDGE_TEMPERATURE );
+    } else if ( freezer_here ){
+        add_msg( "IN_FREEZER" );
+        g->m.apply_in_fridge( *this, FREEZER_TEMPERATURE );
+        add_msg( "Temp: %s", FREEZER_TEMPERATURE );
```

Comment:
You want to merge this will these add_msgs?

Reply:
Of course not - this is pre-cleaning. I mentioned in a comment that debug messages are there to help me pinpoint the problem on compiled instance, then they will be purged. This is still [WIP].